doc/index.md

```diff
@@ -348,72 +364,72 @@ Flushes all buffered data on the socket. Returns true on success. Returns false
 
 
 ### `h1_connection:write_request_line(method, path, httpversion, timeout)` <!-- --> {#http.h1_connection:write_request_line}
 
-Sends an HTTP request to a remote server. Yields until success or `timeout`. If the write fails, returns nil, an error message and an error number.
+Sends an HTTP 1.x request line to a remote server, initiating an HTTP request. Yields until success or `timeout`. If the write fails, returns `nil`, an error message and an error number.
 
 
 ### `h1_connection:write_status_line(httpversion, status_code, reason_phrase, timeout)` <!-- --> {#http.h1_connection:write_status_line}
 
-Sends a status response to the remote connection. Yields until success or `timeout`. If the write fails, the funtion returns nil, an error message and an error number. 
+Sends a status response to the remote connection. Yields until success or `timeout`. If the write fails, the funtion returns `nil`, an error message and an error number. 
 
 
 ### `h1_connection:write_header(k, v, timeout)` <!-- --> {#http.h1_connection:write_header}
 
-Writes a header item to the socket as a key:value string. Yields until success or `timeout`. Returns nil, an error message and an error if the write fails. 
+Writes a header item to the socket as a key:value string. Yields until success or `timeout`. Returns `nil`, an error message and an error if the write fails. 
 
 
 ### `h1_connection:write_headers_done(timeout)` <!-- --> {#http.h1_connection:write_headers_done}
 
-Finalizes the http headers in the socket. Yields until success or `timeout`. Returns nil, an error message and an error if the write fails.
+Terminates a header block by sending "\r\n". Yields until success or `timeout`. Returns `nil`, an error message and an error if the write fails.
 
 
 ### `h1_connection:write_body_chunk(chunk, chunk_ext, timeout)` <!-- --> {#http.h1_connection:write_body_chunk}
  
-Writes a line of data to the socket. `chunk_ext` must be nil as chunck extensions are not supported. Will yield until complete or `timeout` is exceeded. Returns true on success. Returns nil, an error message and an error number if the write fails. 
+Writes a line of data to the socket. `chunk_ext` must be `nil` as chunck extensions are not supported. Will yield until complete or `timeout` is exceeded. Returns true on success. Returns `nil`, an error message and an error number if the write fails. 
 
 
 ### `h1_connection:write_body_last_chunk(chunk_ext, timeout)` <!-- --> {#http.h1_connection:write_body_last_chunk}
 
-Finalizes the message body. `chunck_ext` must be nil as chunk extensions are not supported. Will yield until complete or `timeout` is exceeded. Returns nil, an error message and an error number if the write fails.
+Writes a trailer ("0\r\n") to signal the end of the HTTP body. `chunck_ext` must be `nil` as chunk extensions are not supported. Will yield until complete or `timeout` is exceeded. Returns `nil`, an error message and an error number if the write fails.
```

Comment:
The `"0\r\n"` isn't the trailer (that comes afterwards); it is the separator that signals the end of a chunked body.